frontend/src/components/Quiz/Question.tsx

```diff
@@ -0,0 +1,112 @@
+import { QuestionProp } from "./QuestionProp.type";
+import {
+  Heading,
+  Stack,
+  VStack,
+  useRadioGroup,
+  Box,
+  useRadio,
+  Button,
+} from "@chakra-ui/react";
+import { primaryButtonStyleProps } from "../../utils";
+import { ArrowForwardIcon } from "@chakra-ui/icons";
+import { useQuiz } from "../../contexts/quizContext";
+import {
+  INCREMENT_QUESTION_NUMBER,
+  SELECT_OPTION,
+  SHOW_RESULT,
+  UPDATE_SCORE,
+} from "../../reducers";
+
+function Question(prop: QuestionProp) {
+  const { question: quizQuestion } = prop;
+  const { id: questionId, question, options } = quizQuestion;
+  const {
+    state: { currentQuestionNumber, quizAttempt },
+    dispatch,
+  } = useQuiz();
+
+  const updateSelectedOption = (optionId: string) =>
+    dispatch({ type: SELECT_OPTION, payload: { questionId, optionId } });
+
+  const updateQuestionNumber = () => {
+    quizQuestion.isAttempted &&
+      dispatch({ type: UPDATE_SCORE, payload: { questionId } });
+    currentQuestionNumber >= quizAttempt!.totalQuestions
+      ? dispatch({ type: SHOW_RESULT })
+      : dispatch({ type: INCREMENT_QUESTION_NUMBER });
+  };
+
+  const { getRadioProps, getRootProps } = useRadioGroup({
+    name: "Options",
+    onChange: updateSelectedOption,
+  });
+  const group = getRootProps();
+
+  return (
+    <>
+      <Stack spacing={[6, 8]}>
+        <Heading fontSize={"xl"}>{question}</Heading>
+        <VStack {...group} spacing={[4, 6]}>
+          {options.map((option) => {
+            const { text, id } = option;
+            const radio = getRadioProps({ value: id });
```

Comment:
Avoid initializing and calculating values inside the return function of the main function move it outside of the return statement and consume it. Also, name this file as QuestionCard.tsx instead of Question.tsx

Reply:
Actually, I don't like repeating `element.something` everytime and as I need the whole `element` and it's properties at same time inside `map`, I need to destructure it inside `map()`
Done